brave/brave-core
# Implement crash reporting permission ask dialog

Launch crash reporting permission ask dialog on startup after crash.
Launch tab restore dialog after permission ask dialog is closed.

By overriding, ShowIfNotOffTheRecordProfileBrave() is called instead of
ShowIfNotOffTheRecordProfile().
ShowIfNotOffTheRecordProfileBrave() will launch crash reporting permission
ask dialog and then tab restore dialog is launched after permission ask dialog
is closed.

<img width="449" alt="Screen Shot 2021-07-27 at 9 56 19 PM" src="https://user-images.githubusercontent.com/6786187/127157366-d3d2eb11-3b7b-41a9-bcb5-b57e6165c817.png">
<img width="449" alt="Screen Shot 2021-07-27 at 9 56 38 PM" src="https://user-images.githubusercontent.com/6786187/127157376-9badbdf8-ab52-4625-8ef0-9def1e609bae.png">



Resolves https://github.com/brave/brave-browser/issues/1841

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`npm run test brave_unit_tests -- --filter=MetricsUtilTest*`
`npm run test brave_browser_tests -- --filter=CrashReportPermissionAskDialogTest*`

1. Launch browser and open several tabs
2. Check crash report is not enabled from brave://settings/privacy
3. Load brave://inducebrowsercrashforrealz to simulate browser crash
4. Check crash reporting permission ask dialog is launched
4-1. Click Block and check tab restore bubble is displayed
4-2. Click Allow and check tab restore bubble is displayed
4-2-1. Check crash reporting is enabled from brave://settings/privacy
